packages/code-du-travail-frontend/pages/newsletter.js

```diff
@@ -0,0 +1,243 @@
+    if (/.+@.+\..+/.test(email)) {
+      setStatus(STATUS.INITIAL);
+      return true;
+    }
+    setStatus(STATUS.INVALID);
+    return false;
+  }, []);
+
+  const onSubmit = useCallback(
+    async email => {
+      if (!checkEmail(email, setStatus)) {
+        return;
+      }
+      try {
+        setStatus(STATUS.LOADING);
+        const response = await fetch(
+          `https://sarbacaneapis.com/v1/lists/${SARBACANE_LIST_ID}/contacts/upsert`,
+          {
+            method: "POST",
+            headers: new Headers({
+              accountId: SARBACANE_ACCOUNT_ID
+            }),
+            body: JSON.stringify({ email })
+          }
+        );
+        if (response.ok) {
+          setStatus(STATUS.REGISTERED);
+        } else {
+          setStatus(STATUS.ERRORED);
+          const data = await response.json();
+          Sentry.captureException(data.message);
+        }
+      } catch (error) {
+        setStatus(STATUS.ERRORED);
+        Sentry.captureException(error);
+      }
+    },
+    [checkEmail]
+  );
+
+  let currentView = null;
+  switch (status) {
+    case STATUS.REGISTERED:
+      currentView = (
+        <CenterDiv>
+          <Toast shadow animate="from-top">
+            <StyledHeading as={Heading}>
+              L’enregistrement s’est bien effectué.
+            </StyledHeading>
+            <CenterDiv>
+              Merci de l’intérêt que vous portez au Code du Travail
+              numérique&nbsp;!
+              <br />
+              <Link href="/" passHref>
+                <Button variant="primary" as={StyledLink}>
+                  Retourner sur la page d’accueil <StyledArrowRight />
+                </Button>
+              </Link>
+            </CenterDiv>
+          </Toast>
+        </CenterDiv>
+      );
+      break;
+    case STATUS.ERRORED:
+      currentView = (
+        <CenterDiv>
+          <Toast shadow variant="primary" animate="from-top">
+            <StyledHeading as={Heading}>
+              Oups, une erreur est survenue pendant l’inscription.
+            </StyledHeading>
+            <CenterDiv>
+              Nos équipes ont été informées.
+              <br />
+              <Link href="/" passHref>
+                <Button variant="primary" as={StyledLink}>
+                  Retourner sur la page d’accueil <StyledArrowRight />
+                </Button>
+              </Link>
+            </CenterDiv>
+          </Toast>
+        </CenterDiv>
+      );
+      break;
+    default:
+      currentView = (
+        <Wrapper variant="main">
+          <Form
+            onSubmit={e => {
+              e.preventDefault();
+              onSubmit(email);
+            }}
+          >
+            <Label htmlFor="newsletter-email">Votre adresse email</Label>
+            <Input
+              name="email"
+              id="newsletter-email"
+              invalid={status === STATUS.INVALID}
+              disabled={status === STATUS.LOADING}
+              onChange={event => {
```

Comment:
why not use `useCallback` for event handling

Reply:
Là ça implique de créer un useCallback supplémentaire juste pour ça, c'est peut-etre plus opti (dans le sens ou le useCallback sera moi souvent redéfinit que la fonction anonyme) mais ça va rallonger le code et en terme de perf, à mon avis, c'est négligeable mais si tu préfères je le fais volontiers

Reply:
disons que quitte a mettre un useCallback, autant le mettre sur le onChange que sur le onSubmit